### ISSUE TITLE
Merge DynamoDB plugin into JanusGraph upstream

### ISSUE THREAD

**Issue by amcp** [JanusGraph/janusgraph#1424]
I think 0.2.X support is were merging DynamoDB should start to support Titan 1.0.0 compatibility, and then we can cherry-pick that commit and massage it into the 0.3.X series.

**Comment by amcp** [comment]
@mbrukman

**Comment by FlorianHockmann** [comment]
Shouldn't we have a discussion first on janusgraph-dev of whether we want to add more backends to JanusGraph in general and whether all new backends should be merged into the main repo or in a separate repo?

**Comment by amcp** [comment]
@FlorianHockmann this sounds like a good idea. Do you want to start one or should I? We should be open to contributions in general, but this particular ask has been around for a while. I believe it has been discussed and asked for in one of our mailing lists.

**Comment by FlorianHockmann** [comment]
Yes, we should be open to contributions in general, but a new backend is basically a completely new project that needs to be maintained. If we accept it into JanusGraph, then we also need to maintain it.
I want to avoid a situation where we have a bunch of backends that were donated by contributors who aren't around to maintain them anymore which means more work for the few regular contributors that we currently have.

My point is not directly aimed at the DynamoDB backend, but if we accept that, then why not [a Couchbase backend](https://groups.google.com/forum/#!topic/janusgraph-dev/WcGdLGuqpgs), [Ted's FoundationDB backend](https://github.com/experoinc/janusgraph-foundationdb) or backend XY someone developed just for fun?

If we accept a new backend, then it could also be an option to let it live in its own repo. That way, it doesn't slow down our builds and the development process in general.



I think it makes sense if you start the discussion as you want to merge the backend.

**Comment by porunov** [comment]
I would agree here with @FlorianHockmann . 

Also, I think it would be better when all backends would be separate projects. I think they could be hosted in JanusGraph github as separate repositories if they are maintained by someone. Generally, we could have "officially supported backends" and "third party backends" (which are hosted not in JanusGraph github).

I agree that discussion is needed.
